File: _prog/_java/_old/_train/_poo/Conta.java
/**
 * Conta
 */
public class Conta extends Cliente{

    public String titular;
    private double saldo;
    public double limite;
    protected Cliente cliente = new Cliente();

    public boolean depositarEm(double valor){
        if(valor >= 0){
            this.saldo += valor;
            return true;
        }
        return false;
    }

    public boolean sacarDinheiro(double valor){
        if( valor <= 0 || valor > this.saldo )
            return false;
        else
            this.saldo -= valor;
            return true;
    }

    public boolean tresferirPara(double valor, Conta destino){
        if(valor > this.saldo){
            return false;
        }
        destino.saldo += valor;
        this.saldo -= valor;
        return true;
    }

    @Override
    public String toString() {
        return "Conta [limite=" + limite + ", saldo=" + saldo + ", titular=" + titular + "]";
    }

    
}

File: _prog/_python/_aulas/_arquivos/conta.py
class Cliente:
    def __init__(self, nome, sobrenome, numBi):
        self.nome = nome
        self.sobrenome = sobrenome
        self.numBi = numBi


class Conta:
    def __init__(self, numero, cliente, saldo, limite):

        print('Inicializando uma conta!')

        self.numero = numero
        self.cliente = cliente
        self.saldo = saldo
        self.limite = limite

    def criar(numero, titular, saldo, limite):
        conta = {'numero':numero, 'titular':titular, 'saldo':saldo, 'limite':limite}
        return conta

    def depositar(self, valor):
        self.saldo += valor

    def sacar(self, valor):
        if self.saldo < valor:
            return False
        else:
            self.saldo -= valor
            return True

    def extrato(self):
        print('numero: {}\nsaldo: {}'.format(self.numero, self.saldo))

    def transferir(self, destino, valor):
        retirou = self.sacar(valor)
        if retirou == False:
            return False
        else:
            destino.saldo += retirou
            return True

